File: Algothrim/project_euler/3/3.cpp
#include <iostream>

int main(void)
{
    unsigned long num = 600851475143;
    unsigned long smallest_prime_factor = 2;

    while(num % smallest_prime_factor != 0) {
        smallest_prime_factor++;
    }

    std::cout << "largest_prime_factor is: "
        << num / smallest_prime_factor << std::endl;

    return 0;
}

File: Algothrim/project_euler/3/3.go
package main

import (
	"fmt"
)

func main() {
	var num uint64 = 600851475143
	var smallest_prime_factor uint64 = 2

	for num%smallest_prime_factor != 0 {
		smallest_prime_factor++
	}

	fmt.Println("largest_prime_factor is:", num/smallest_prime_factor)
}
